src/MergeContext.js

```diff
@@ -1229,9 +1238,17 @@ class PullRequest {
          * Until _loadRawPr(), we must avoid this._rawPr fields except .number.
          * TODO: Refactor to eliminate the risk of too-early this._rawPr use.
          */
+
+        await this._loadLabels();
+
+        // We obtained labels and can cleanup now.
+        // In a case of a new error, fresh values for error labels will be calculated.
+        this._labels.remove(Config.failedOtherLabel());
```

Comment:
> Alex: Perhaps we should call _removeTemporaryLabels() here

> Eduard: No, we cannot: for example, we should leave the 'invalid message' label as is until we are able to check it again.

I am not sure I am ready to buy your argument. In all other cases, we call _removeTemporaryLabels() essentially because we are about to compute the right set of fresh labels. All previous temporary labels are potentially stale and should be removed so that they are not "leaked" for humans to see. The situation is exactly the same here AFAICT!

Let's consider your specific example -- the failedDescriptionLabel(). Let's assume we remove it here unconditionally. I see three possible outcomes:

* If we crash before finishing PR processing, nothing will change. The old (possibly stale) label will stay on GitHub. That is the best outcome we can hope for under the circumstances.
* If we orderly terminate PR processing before checking the current PR description, then the (potentially stale) label will be removed from GitHub. I think this is the best outcome given the current code limitations: Given the two choices, it is better to clear the still-fresh label than to risk preserving the already-stale one, especially because there will be another label that would indicate a more serious problem. There is an "If multiple failures need labeling, label all of them" TODO that would reduce the number of cases where there are only two (bad) choices in this scenario.
* If we orderly terminate PR processing after checking the current PR description, then the GitHub label will reflect the current PR state, which is the best outcome possible.

We also never check whether failedDescriptionLabel() is set.

I think the same logic can apply to other temporary labels. What do you think?

```suggestion
        // methods below compute fresh labels from scratch without worrying
        // about stale labels, so we clear all the labels that we must sync
        this._removeTemporaryLabels();
```

Reply:
Agreed and done at b160b99.